extension/src/extension.ts

```diff
@@ -122,7 +123,7 @@ async function runTest(javaHome: string, tests: TestSuite[] | TestSuite, storage
     const suites = testList.map((s) => s.test);
     const uri = vscode.Uri.parse(testList[0].uri);
     const classpaths = classPathManager.getClassPath(uri);
-    const port = readPortConfig();
+    const port = await Utility.getFreePort();
```

Comment:
try ... catch for getFreePort to handle the failure case.

Reply:
+1